File: backend/default/responses/playlistResponse.py
from typing import List, Optional
from pydantic import BaseModel


class PlaylistMediaResponse(BaseModel):
    id: int
    position: int
    media_type: str
    media_id: str
    provider_id: Optional[int] = None


class PlaylistContributorResponse(BaseModel):
    user_id: int
    role_key: int


class PlaylistResponse(BaseModel):
    id: int
    public_id: str
    name: str
    description: Optional[str] = None
    cover_image: str
    is_public: bool
    owner_id: int
    date_added: str
    date_updated: str
    medias: List[PlaylistMediaResponse] = []
    contributors: List[PlaylistContributorResponse] = []


class PlaylistListResponse(BaseModel):
    playlists: List[PlaylistResponse]

File: frontend/dto/playlistResponse.ts
import {
    PlaylistContributorResponseSchema,
    PlaylistMediaResponseSchema,
} from "@/dto";
import { z } from "zod";

export const PlaylistResponseSchema = z.object({
    id: z.number(),
    public_id: z.string(),
    name: z.string(),
    description: z.string().nullable(),
    cover_image: z.string(),
    is_public: z.boolean(),
    owner_id: z.number(),
    date_added: z.string(),
    date_updated: z.string(),
    medias: z.array(z.lazy(() => PlaylistMediaResponseSchema)),
    contributors: z.array(z.lazy(() => PlaylistContributorResponseSchema)),
});

export type PlaylistResponse = z.infer<typeof PlaylistResponseSchema>;
